event/event.go

```diff
@@ -110,3 +116,25 @@ type Subscription struct {
 func (s *Subscription) Message() <-chan Message {
 	return s.value
 }
+
+// P2PTopic is an event that is published when a peer has updated the topics it is subscribed to.
+type P2PTopic struct {
+	ToAdd    []string
+	ToRemove []string
+}
+
+// PeerInfo is an event that is published when the node has updated its peer info.
+type PeerInfo struct {
+	Info peer.AddrInfo
+}
+
+// Replicator is an event that is published when a replicator is added or updated.
+type Replicator struct {
+	// The peer info for the replicator instance.
+	Info peer.AddrInfo
+	// The map of schema roots that the replicator will receive updates for.
+	Schemas map[string]struct{}
+	// Docs will receive Updates if new collections have been added to the replicator
+	// and those collections have documents to be replicated.
+	Docs <-chan Update
+}
```

Comment:
this just means that we have a problem with design and code structure. Moving stuff here is just another work-around.
2 packages can not be dependent of each other, that means whoever is independent can host the event definition.
But it's your call. I believe this PR is a good step towards cleaner code structure overall and I'm fine with keep the events here as a temporary solution. Would be nice to ticket though to keep track of our technical dept.

Reply:
I'm fine with having the event types in a different package than `event`. I just don't think it's a good approach to have them spread around in their "most appropriate" package for the reason I stated above and more.

`core` is internal and the `bus` is public. Maybe `client` would be better?

Reply:
I'm happy with `client`, although a pub-internal split might occur with event types this way (I'm happy with that, but you might not be?)

Reply:
There would be no split. Everything would be in `client` in this scenario. What I want to avoid is creating dependencies with packages that are one-of-possibly-many/interchangeable. The goal of this PR is to remove the dependencies. I don't want to recreate it with event messages.

Reply:
Just to clarify, when I say everything I mean every event types. Not the bus implementation itself.

Reply:
You don't picture us ever needing/wanting internal event types?

Reply:
If you mean that the public ones can be in `client/event` and the internal ones in `internal/core/event` or something like that then yes I think that's totally fine and probably desired.

Reply:
Nice, Islam are you happy with this?

And if so, are you happy with re-orging the events into `client`/`core` in another PR (if that's what Fred prefers)?  I do see moving them as out of scope here, and if I was the author would rather do it elsewhere.

Reply:
yeah, I'm fine with, also with doing this in a separate PR.